src/main/java/org/example/bookstore/repository/book/spec/TitleSpecificationProvider.java

```diff
@@ -0,0 +1,22 @@
+package org.example.bookstore.repository.book.spec;
+
+import java.util.Arrays;
+import org.example.bookstore.model.Book;
+import org.example.bookstore.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TitleSpecificationProvider implements SpecificationProvider<Book> {
+    private static final String TITLE = "title";
```

Comment:
I would reuse this constant in the builder class as well instead of creating a duplicate, same for author